dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,546 @@
+    >>> from sklearn.linear_model import SGDClassifier
+    >>>
+    >>> X, y = make_classification(chunks=20)
+    >>> est = SGDClassifier(tol=1e-3)
+    >>> params = {'alpha': np.logspace(-4, 0, num=1000),
+    >>>           'loss': ['hinge', 'log', 'modified_huber', 'squared_hinge'],
+    >>>           'average': [True, False]}
+    >>>
+    >>> search = HyperbandCV(est, params)
+    >>> search.fit(X, y, classes=np.unique(y))
+    >>> search.best_params_
+    ... {'loss': 'log', 'average': False, 'alpha': 0.0080502}
+
+    Attributes
+    ----------
+    cv_results_ : dict of lists
+        Information about the cross validation scores for each model.
+        All lists are ordered the same, and this value can be imported into
+        a pandas DataFrame. This dict has keys of
+
+            * ``rank_test_score``
+            * ``model_id``
+            * ``mean_fit_time``
+            * ``mean_score_time``
+            * ``std_fit_time``
+            * ``std_score_time``
+            * ``mean_test_score``
+            * ``std_test_score``
+            * ``partial_fit_calls``
+            * ``mean_train_score``
+            * ``std_train_score``
+            * ``params``
+            * ``param_value``
+            * ``mean_copy_time``
+
+    meta_ : dict
+        Information about every model that was trained. Can be used as input to
+        ``fit_metadata``.
+    history_ : list of dicts
+        Information about every model after it is scored. Most models will be
+        in here more than once because poor performing models are "killed" off
+        early.
+    best_params_ : dict
+        The params that produced the best performing model
+    best_estimator_ : any
+        The best performing model
+    best_index_ :
+    n_splits_ : int
+        The number of cross-validation splits.
+    multimetric_ :
+        Whether or whether this model selection algorithm is multimetric.
+    test_size : float, default 0.15
+        The fraction of test set that should be used for testing.
+
+    Notes
+    -----
+    Hyperband is state of the art via an adaptive scheme: that only spends time
+    on high-performing models, because our goal is to find the highest
+    performing model. This means that it stops training models that perform
+    poorly.
+
+    References
+    ----------
+    1. "Hyperband: A novel bandit-based approach to hyperparameter
+       optimization", 2016 by L. Li, K. Jamieson, G. DeSalvo, A. Rostamizadeh,
+       and A. Talwalkar.  https://arxiv.org/abs/1603.06560
+
+    """
+    def __init__(self, model, params, max_iter=81, eta=3, asynchronous=None,
+                 random_state=42, scoring=None, test_size=0.15):
+        self.model = model
+        self.params = params
+        self.max_iter = max_iter
+        self.eta = eta
+        self.test_size = test_size
+        self.random_state = random_state
+        if asynchronous is None:
+            asynchronous = True
```

Comment:
It looks like the default behavior is for `asynchronous=True`.  We should place this into the parameter list directly rather than encode it in logic in the constructor.

Reply:
I've changed the default to `asynchronous=True`.

I left the default as `asynchronous=None` for future work in fear of deprecation. I am planning some future work that would involve making some smart choice between `True` and `False` based off `Client.ncores` and `max_iter`. This won't be complicated, and I don't think it'll take too long.

Reply:
By setting this value in the constructor we're already making this choice.  There is no behavioral change in setting this in the constructor or as a default in the parameters